From 9745863ec676d79d6a65f050fdf7bf7654c9d0ae Mon Sep 17 00:00:00 2001
From: holishing <holishingtest@gmail.com>
Date: Tue, 21 Nov 2017 21:22:40 +0800
Subject: [PATCH] Try running in 64-bit environment (#1)

still in 32-bit structure
How to compile: https://stackoverflow.com/questions/23638271
Thanks to lantw44.bbs<at>sony.tfcis.org 's patch
---
 daemon/Makefile      | 2 +-
 game/Makefile        | 4 ++--
 innbbsd/Makefile     | 2 +-
 lib/Makefile         | 2 +-
 maple/Makefile       | 2 +-
 pip/Makefile         | 4 ++--
 so/Makefile          | 4 ++--
 util/Makefile        | 2 +-
 util/backup/Makefile | 2 +-
 util/tran/Makefile   | 2 +-
 util/uno/Makefile    | 2 +-
 11 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/daemon/Makefile b/daemon/Makefile
index ea32c60..88d1a15 100644
--- a/daemon/Makefile
+++ b/daemon/Makefile
@@ -31,7 +31,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" LDFLAGS="-s -L../lib -ldao" $(EXE)
 
 linux:
-	@$(MAKE) CFLAGS="-DLINUX -O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../lib -ldao -lcrypt -lresolv" $(EXE)
+	@$(MAKE) CFLAGS="-m32 -DLINUX -O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../lib -ldao -lcrypt -lresolv" $(EXE)
 
 solaris:
 	@$(MAKE) CFLAGS="-DSOLARIS -O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../lib -ldao -lsocket -lresolv -lnsl -L/usr/ucblib -lucb -R/usr/ucblib" $(EXE)
diff --git a/game/Makefile b/game/Makefile
index fde06e8..c141f02 100644
--- a/game/Makefile
+++ b/game/Makefile
@@ -24,7 +24,7 @@ SO =	bar.so bingo.so bj.so bwboard.so chessmj.so dice.so dragon.so fantan.so \
 .SUFFIXES: .c .o .so
 
 .c.o:	;  $(CC) $(CFLAGS) -c $*.c
-.o.so:	;  ld -s -G $*.o -o $*.so -L../lib -ldao
+.o.so:	;  ld -s -G $*.o -o $*.so -L../lib -ldao -melf_i386
 
 
 all: 
@@ -41,7 +41,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
 
 linux:
-	@$(MAKE) CFLAGS="-DLINUX -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
+	@$(MAKE) CFLAGS="-m32 -DLINUX -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
 
 solaris:
 	@$(MAKE) CFLAGS="-DSOLARIS -DSYSV -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
diff --git a/innbbsd/Makefile b/innbbsd/Makefile
index adcada6..1eac6fd 100644
--- a/innbbsd/Makefile
+++ b/innbbsd/Makefile
@@ -38,7 +38,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" LDFLAGS="-s -L../lib -ldao" $(EXE)
 
 linux:
-	@$(MAKE) CFLAGS="-O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../lib -ldao" $(EXE)
+	@$(MAKE) CFLAGS="-m32 -O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-m32 -s -L../lib -ldao" $(EXE)
 
 solaris:
 	@$(MAKE) CFLAGS="-DSOLARIS -O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../lib -ldao -lsocket -lnsl -L/usr/ucblib -lucb" $(EXE)
diff --git a/lib/Makefile b/lib/Makefile
index 1578771..4dbd1aa 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -11,7 +11,7 @@
 CC	= gcc
 RANLIB	= ranlib
 CPROTO	= cproto -E"gcc -pipe -E" -I../include # -s -v
-CFLAGS	= -O2 -pipe -fomit-frame-pointer -Wunused -I../include
+CFLAGS	= -m32 -O2 -pipe -fomit-frame-pointer -Wunused -I../include
 
 
 # ------------------------------------------------------ #
diff --git a/maple/Makefile b/maple/Makefile
index be8427c..cc37cad 100644
--- a/maple/Makefile
+++ b/maple/Makefile
@@ -39,7 +39,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" LDFLAGS="-s -L../lib -ldao" $(EXE)
 
 linux:
-	@$(MAKE) CFLAGS="-DLINUX -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" LDFLAGS="-s -L../lib -ldao -lcrypt -lresolv -ldl -rdynamic" $(EXE)
+	@$(MAKE) CFLAGS="-m32 -DLINUX -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" LDFLAGS="-m32 -s -L../lib -ldao -lcrypt -lresolv -ldl -rdynamic" $(EXE)
 
 solaris:
 	@$(MAKE) CFLAGS="-DSOLARIS -DSYSV -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" LDFLAGS="-s -L../lib -ldao -ltermcap -lsocket -lnsl -lresolv -lelf -ldl" $(EXE)
diff --git a/pip/Makefile b/pip/Makefile
index 71c64d6..15c9320 100644
--- a/pip/Makefile
+++ b/pip/Makefile
@@ -37,7 +37,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
 
 linux:
-	@$(MAKE) CFLAGS="-DLINUX -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
+	@$(MAKE) CFLAGS="-m32 -DLINUX -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
 
 solaris:
 	@$(MAKE) CFLAGS="-DSOLARIS -DSYSV -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
@@ -56,7 +56,7 @@ cygwin:
 
 
 pip.so: $(OBJ)
-	ld -s -G $(OBJ) -o pip.so -L../lib -ldao
+	ld -s -G $(OBJ) -o pip.so -L../lib -ldao -melf_i386
 
 
 install: $(SO)
diff --git a/so/Makefile b/so/Makefile
index 5963383..d8f1fc9 100644
--- a/so/Makefile
+++ b/so/Makefile
@@ -24,7 +24,7 @@ SO =	admutil.so aloha.so bank.so chat.so help.so innbbs.so manage.so newbrd.so \
 .SUFFIXES: .c .o .so
 
 .c.o:	;  $(CC) $(CFLAGS) -c $*.c
-.o.so:	;  ld -s -G $*.o -o $*.so -L../lib -ldao
+.o.so:	;  ld -s -G $*.o -o $*.so -L../lib -ldao -melf_i386
 
 
 all: 
@@ -41,7 +41,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
 
 linux:
-	@$(MAKE) CFLAGS="-DLINUX -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
+	@$(MAKE) CFLAGS="-m32 -DLINUX -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
 
 solaris:
 	@$(MAKE) CFLAGS="-DSOLARIS -DSYSV -O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" $(SO)
diff --git a/util/Makefile b/util/Makefile
index 8271ca4..9cc6b25 100644
--- a/util/Makefile
+++ b/util/Makefile
@@ -33,7 +33,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../include" LDFLAGS="-s -L../lib -ldao" $(EXE)
 
 linux:
-	@$(MAKE) CFLAGS="-O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../lib -ldao -lcrypt -lresolv" $(EXE)
+	@$(MAKE) CFLAGS="-m32 -O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-m32 -s -L../lib -ldao -lcrypt -lresolv" $(EXE)
 
 solaris:
 	@$(MAKE) CFLAGS="-O2 -pipe -I../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../lib -ldao -lsocket -lresolv -lnsl -L/usr/ucblib -lucb -R/usr/ucblib" $(EXE)
diff --git a/util/backup/Makefile b/util/backup/Makefile
index c693a78..971bd39 100644
--- a/util/backup/Makefile
+++ b/util/backup/Makefile
@@ -31,7 +31,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../../include" LDFLAGS="-s -L../../lib -ldao" $(EXE)
 
 linux:
-	@$(MAKE) CFLAGS="-O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../../lib -ldao" $(EXE)
+	@$(MAKE) CFLAGS="-m32 -O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-m32 -s -L../../lib -ldao" $(EXE)
 
 solaris:
 	@$(MAKE) CFLAGS="-O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../../lib -ldao -lsocket -lnsl -L/usr/ucblib -lucb" $(EXE)
diff --git a/util/tran/Makefile b/util/tran/Makefile
index 4702c07..c591d8c 100644
--- a/util/tran/Makefile
+++ b/util/tran/Makefile
@@ -38,7 +38,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../../include" LDFLAGS="-s -L../../lib -ldao" $(EXE)
 
 linux:
-	@$(MAKE) CFLAGS="-O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../../lib -ldao" $(EXE)
+	@$(MAKE) CFLAGS="-m32 -O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-m32 -s -L../../lib -ldao" $(EXE)
 
 solaris:
 	@$(MAKE) CFLAGS="-O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../../lib -ldao -lsocket -lnsl -L/usr/ucblib -lucb" $(EXE)
diff --git a/util/uno/Makefile b/util/uno/Makefile
index 818c9aa..9d32009 100644
--- a/util/uno/Makefile
+++ b/util/uno/Makefile
@@ -31,7 +31,7 @@ sun:
 	@$(MAKE) CFLAGS="-O2 -pipe -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses -I../../include" LDFLAGS="-s -L../../lib -ldao" $(EXE)
 
 linux:
-	@$(MAKE) CFLAGS="-O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../../lib -ldao" $(EXE)
+	@$(MAKE) CFLAGS="-m32 -O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-m32 -s -L../../lib -ldao" $(EXE)
 
 solaris:
 	@$(MAKE) CFLAGS="-O2 -pipe -I../../include -fomit-frame-pointer -Wunused -Wno-unused-result -Wno-parentheses" LDFLAGS="-s -L../../lib -ldao -lsocket -lnsl -L/usr/ucblib -lucb" $(EXE)
